postgres/datadog_checks/postgres/connections.py

```diff
@@ -85,53 +83,54 @@ def __init__(self, check: AgentCheck, connect_fn: Callable[[str, int, int], None
                 )
         self.connect_fn = connect_fn
 
-    def _get_connection_pool(
+    def _get_connection_raw(
         self,
         dbname: str,
         ttl_ms: int,
         timeout: int = None,
-        min_pool_size: int = 1,
-        max_pool_size: int = None,
-        startup_fn: Callable[[ConnectionPool], None] = None,
+        startup_fn: Callable[[psycopg.Connection], None] = None,
         persistent: bool = False,
-    ) -> ConnectionPool:
+    ) -> psycopg.Connection:
         """
         Return a connection pool for the requested database from the managed pool.
         Pass a function to startup_func if there is an action needed with the connection
         when re-establishing it.
         """
+        if not timeout:
+            # We don't want the wait indefinitely for the connection pool to free up
+            # if timeout is not specified, so we set it to the default connection timeout
+            timeout = self._config.connection_timeout
         start = datetime.datetime.now()
         self.prune_connections()
-        with self._mu:
-            conn = self._conns.pop(dbname, ConnectionInfo(None, None, None, None, None))
-            db_pool = conn.connection
-            if db_pool is None or db_pool.closed:
-                if self.max_conns is not None:
-                    # try to free space until we succeed
-                    while len(self._conns) >= self.max_conns:
-                        self.prune_connections()
-                        self.evict_lru()
-                        if timeout is not None and (datetime.datetime.now() - start).total_seconds() > timeout:
-                            raise ConnectionPoolFullError(self.max_conns, timeout)
-                        time.sleep(0.01)
-                        continue
-                self._stats.connection_opened += 1
-                db_pool = self.connect_fn(dbname, min_pool_size, max_pool_size)
-                if startup_fn:
-                    startup_fn(db_pool)
-            else:
-                # if already in pool, retain persistence status
-                persistent = conn.persistent
-
-            deadline = datetime.datetime.now() + datetime.timedelta(milliseconds=ttl_ms)
-            self._conns[dbname] = ConnectionInfo(
-                connection=db_pool,
-                deadline=deadline,
-                active=True,
-                last_accessed=datetime.datetime.now(),
-                persistent=persistent,
-            )
-            return db_pool
+        conn = self._conns.pop(dbname, ConnectionInfo(None, None, None, None, None))
+        db = conn.connection
+        if db is None or db.closed:
+            if self.max_conns is not None:
```

Comment:
i think this max conns logic needs to be re-worked a bit now that we are instantiating one of these per thread... Does it make sense to get rid of this part of the feature for now? 

I think the issue would mainly manifest in the explain plan code path and the schema collection/ autodiscovery code paths, which would hold open a connection to every database it:

1. sees queries running so it can explain them 
2. discovered through autodiscovery and we want to grab relation metrics/schema stuff from

Reply:
It's a bit tricky to maintain a true `max conns` for all threads with this change. I left this untouched for now just to minimize the changes.